File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        result = ""


        for i in range(len(s)):
            l = i
            r = i

            # odd
            while l >= 0 and r < len(s) and s[l] == s[r]:
                if (r - l + 1) > len(result):
                    result = s[l:r+1]
                l -= 1
                r += 1

            # even
            l = i
            r = i + 1
            while l >= 0 and r < len(s) and s[l] == s[r]:
                if (r - l + 1) > len(result):
                    result = s[l:r+1]
                l -= 1
                r += 1

        return result

        
        

# ababd  ---> odd number

# s[0] as center:
# a

# s[1] as center:
# aba

# s[2] as center:
# bab

# d[3] as center:
# d

# abbc ---> even number

File: Data Structures & Algorithms/reverse-bits/submission-0.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t result = 0;
        for (int i = 0; i < 32; i++) {
            result |= (((n >> i) & 1) << (31 - i));
        }

        return result;
    }
};

// 0000 0010 0110

// 000000000001

// 0110 0100 0000